### PR TITLE
drivers: usb: device: stm32: Configure USB Prescaler for F103

### LINKED ISSUE #47146: STM32F103:  USB clock prescaler isn't set during USB initialisation
Whilst the driver for the ST USB sets the USB clock prescaler for the OTG equipped STM32F105/7, the prescaler is left set to it's default with the STM32F103's USB interface. 

This fortunately works correctly when configuring the STM32F103 with a 72MHz SYSCLK from the PLL when nothing else has touched the USB prescaler,, but will likely not work with a 48MHz PLL clock output as the USB prescaler will be incorrectly set (unless something fixes this elsewhere).

There is also a device tree property (`usbpre` on `st,stm32f1-pll-clock`) that looks like it should be used to control the USB prescaler, but it is not actually rigged to anything, nor is there any documentation to explain what the correct values are.  (the only two settings for the F103 USB prescaler are /1 and /1.5, so it's not clear how a `int` property should be set).

**To Reproduce**
Theoretical defect found whilst trying to work out how the USB clock was configured.

existing clock configuration was discovered in `drivers/usb/device/usb_dc_stm32.c`,
and the `usbpre` property for `st,stm32f1-pll-clock` was found to be unused.

**Expected behavior**
That either `drivers/usb/device/usb_dc_stm32.c` sets the clocks correctly for the F103 as it does for the other F10x series devices, or the `usbpre` property is documented in a way that makes sense given the limitations of the USB prescaler in the F103.

**Impact**
annoyance for me personally, but potentially a showstopper for other users of the F103.

### PR DESCRIPTION
**Pull request by kuroneko**
- If the HAL headers expose the USBPRE flag, then we're probably
  dealing with a F103 - using the same fundamental logic as the
  code for the OTG models, set-up the USB Prescaler correctly.

- Fixes #47146

### CODE REVIEW

**Review by erwango — DISMISSED** [review]
I theory I would push to use clock_control  API to switch clock source, but F1 is one of the series where this is not yet available, so no reason to bother.
Approved

File: drivers/usb/device/usb_dc_stm32.c [hunk 1]
```diff
@@ -293,8 +293,28 @@ static int usb_dc_stm32_clock_enable(void)
...
+	/* on other STM32F1 family SOCs, we have a simple /1 or /1.5 divider on
+	 * the back of the RCC.  Similar strategy to the above, but we use the
+	 * correct flags
+	 */
+	if (LL_RCC_GetSysClkSource() == LL_RCC_SYS_CLKSOURCE_STATUS_PLL) {
+		switch (sys_clock_hw_cycles_per_sec()) {
+		case 48000000U:
+			LL_RCC_SetUSBClockSource(LL_RCC_USB_CLKSOURCE_PLL);
+			break;
+		case 72000000U:
```
Comment (FRASTM) [inline drivers/usb/device/usb_dc_stm32.c:306]: case MHZ(72):

File: drivers/usb/device/usb_dc_stm32.c [hunk 2]
```diff
@@ -293,8 +293,28 @@ static int usb_dc_stm32_clock_enable(void)
...
-
-#endif /* RCC_HSI48_SUPPORT / LL_RCC_USB_CLKSOURCE_NONE / RCC_CFGR_OTGFSPRE */
+#elif defined(RCC_CFGR_USBPRE)
+	/* on other STM32F1 family SOCs, we have a simple /1 or /1.5 divider on
+	 * the back of the RCC.  Similar strategy to the above, but we use the
+	 * correct flags
+	 */
+	if (LL_RCC_GetSysClkSource() == LL_RCC_SYS_CLKSOURCE_STATUS_PLL) {
+		switch (sys_clock_hw_cycles_per_sec()) {
+		case 48000000U:
```
Comment (FRASTM) [inline drivers/usb/device/usb_dc_stm32.c:303]: could be written as `case MHZ(48):`

File: drivers/usb/device/usb_dc_stm32.c (hunk 1 above)
Comment (kuroneko) [reply drivers/usb/device/usb_dc_stm32.c]: Fixed this, as well as the original instance in the source in the OTG initialisation.

File: drivers/usb/device/usb_dc_stm32.c (hunk 2 above)
Comment (kuroneko) [reply drivers/usb/device/usb_dc_stm32.c]: Fixed this, as well as the original instance in the source in the OTG initialisation.

**Review by erwango — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**